gaphor/diagram/diagramtools/placement.py

```diff
@@ -0,0 +1,216 @@
+import logging
+from typing import Callable, Optional, Type, TypeVar
+
+from gaphas.aspect.connector import Connector as ConnectorAspect
+from gaphas.aspect.connector import ItemConnector
+from gaphas.aspect.handlemove import HandleMove
+from gaphas.tool.placement import PlacementState
+from gaphas.view import GtkView
+from gi.repository import Gtk
+
+from gaphor.core import transactional
+from gaphor.core.modeling import Diagram, Element
+from gaphor.diagram.connectors import Connector
+from gaphor.diagram.event import DiagramItemPlaced
+from gaphor.diagram.grouping import Group
+from gaphor.diagram.presentation import Presentation
+
+log = logging.getLogger(__name__)
+
+P = TypeVar("P", bound=Presentation, covariant=True)
+FactoryType = Callable[[Diagram, Optional[Presentation]], Presentation]
+ConfigFuncType = Callable[[P], None]
+
+
+def placement_tool(
+    view: GtkView, factory: FactoryType, event_manager, handle_index: int
+):
+    gesture = Gtk.GestureDrag.new(view)
+    placement_state = PlacementState(factory, handle_index)
+    gesture.connect("drag-begin", on_drag_begin, placement_state)
+    gesture.connect("drag-update", on_drag_update, placement_state)
+    gesture.connect("drag-end", on_drag_end, placement_state, event_manager)
+    return gesture
+
+
+def on_drag_begin(gesture, start_x, start_y, placement_state):
```

Comment:
I noticed some of the new code doesn't have type hints, did we want to add those in before merging?